reana_client/validation.py

```diff
@@ -637,21 +656,33 @@ def parse_params(stages):
     _warn_not_defined_parameters(cmd_param_steps_mapping, workflow_params, "yadage")
 
 
-def _validate_dangerous_operations(command, step):
+def _validate_dangerous_operations(command, step, step_flag=None):
     """Warn the user if a command has dangerous operations.
 
     :param command: A workflow step command to validate.
     :param step: The workflow step that contains the given command.
+    :param step_flag: True if workflow has `steps` param else None
     """
+
+    def _display_message(operation, step, step_flag):
+        msg = 'Operation "{}" found in step "{}" might be dangerous.'
+        if not step_flag:
+            msg = 'Operation "{}" might be dangerous.'
+        display_message(
+            msg.format(operation.strip(), step if step_flag else None),
+            msg_type="warning",
+            indented=True,
+        )
+
     for operation in COMMAND_DANGEROUS_OPERATIONS:
-        if operation in command:
-            display_message(
-                'Operation "{}" found in step "{}" might be dangerous.'.format(
-                    operation.strip(), step
-                ),
-                msg_type="warning",
-                indented=True,
-            )
+        if isinstance(command, str) and operation in command:
+            _display_message(operation, step, step_flag)
+        if isinstance(command, list):
```

Comment:
A command should never be a list, could we move this logic to the `_check_dangerous_operations` inside CWL and leave this function more generic?